src/Controller/Component/PaginatorComponent.php

```diff
@@ -81,6 +81,8 @@ public function __construct(ComponentRegistry $registry, array $config = [])
             $this->_paginator = new Paginator();
         }
 
+        // Call parent setConfig() to merge with defaults before proxy call to Paginator
+        parent::setConfig($config);
```

Comment:
This isn't enough. The merged config isn't passed to PaginatorComponent::setConfig() in the constructor, the original config is.